File: scripts/sync_versions.py
#!/usr/bin/env python3
"""Sync version across all project files."""

import json
import sys
from pathlib import Path


def update_vscode_extension_version(version: str) -> None:
    """Update VS Code extension package.json version."""
    package_json_path = Path("vscode-extension/package.json")
    if not package_json_path.exists():
        print(f"Warning: {package_json_path} not found")
        return

    with open(package_json_path, encoding="utf-8") as f:
        package_data = json.load(f)

    package_data["version"] = version

    with open(package_json_path, "w", encoding="utf-8") as f:
        json.dump(package_data, f, indent=2)
        f.write("\n")

    print(f"✅ Updated VS Code extension version to {version}")


def main() -> None:
    """Main version sync function."""
    if len(sys.argv) != 2:
        print("Usage: python sync_versions.py <version>")
        sys.exit(1)

    version = sys.argv[1]
    print(f"🔄 Syncing version to {version}")

    # Update VS Code extension
    update_vscode_extension_version(version)

    print(f"✅ Version sync complete: {version}")


if __name__ == "__main__":
    main()

File: scripts/syncVersions.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');

// Get the new version from command line argument
const newVersion = process.argv[2];

if (!newVersion) {
  console.error('Error: No version provided');
  process.exit(1);
}

console.log(`Syncing version to ${newVersion}`);

// Update root package.json
const rootPackagePath = path.join(__dirname, '..', 'package.json');
const rootPackage = JSON.parse(fs.readFileSync(rootPackagePath, 'utf8'));
rootPackage.version = newVersion;
fs.writeFileSync(rootPackagePath, JSON.stringify(rootPackage, null, 2) + '\n');
console.log(`Updated root package.json to version ${newVersion}`);

// Update VS Code extension package.json
const extensionPackagePath = path.join(__dirname, '..', 'vscode-extension', 'package.json');
const extensionPackage = JSON.parse(fs.readFileSync(extensionPackagePath, 'utf8'));
extensionPackage.version = newVersion;
fs.writeFileSync(extensionPackagePath, JSON.stringify(extensionPackage, null, 2) + '\n');
console.log(`Updated VS Code extension package.json to version ${newVersion}`);

console.log('Version sync complete!');
